### PR TITLE
Summary.ipf

### PR DESCRIPTION
**Pull request by alexkowa**
There is a new function to summarize ipf output and I put it as S3 method

### CODE REVIEW

**Comment by GregorDeCillia** [comment]
I can't get this to work

```r
remotes::install_github("statistikat/surveysd@925461f1a942f054be12d67068c0db59d3097bfe")
library(surveysd)
example("summary.ipf", "surveysd", run.dontrun = TRUE)
#> Error in summary.ipf(calibweights1) : 
#>    could not find function "summary.ipf"
eusilc <- demo.eusilc(n = 1, prettyNames = TRUE)
conP1 <- xtabs(pWeight ~ age, data = eusilc)
conP2 <- xtabs(pWeight ~ gender + region, data = eusilc)
conP3 <- xtabs(pWeight * eqIncome ~ gender, data = eusilc)
conH1 <- xtabs(pWeight ~ hsize + region, data = eusilc)
calibweights1 <- ipf(eusilc, conP = list(conP1, conP2, eqIncome = conP3), bound = NULL)
summary(calibweights1)
#> Error in get(w) : invalid first argument
```

According to `traceback()`, the error is thrown from here https://github.com/statistikat/surveysd/blob/925461f1a942f054be12d67068c0db59d3097bfe/R/summary.ipf.R#L88-L91

The attribute `baseWeight` is missing but required in `summary`

```r
av <- attributes(calibweights1)
(w <- av$baseweight)
#> NULL
get(w)
#> Error in get(w) : invalid first argument
```

It seems that the baseweight hass a value of `NULL` when it is set here https://github.com/statistikat/surveysd/blob/925461f1a942f054be12d67068c0db59d3097bfe/R/ipf.r#L415

**Comment by alexkowa** [comment]
ah yes, thank you. It only works with ipf calls with a baseweight provided. I will fix that.

**Comment by GregorDeCillia** [comment]
There is another issue with an undefined variable 

```r
eusilc$w_base <- 1
calibweights1 <- ipf(eusilc, conP = list(conP1, conP2, eqIncome = conP3), bound = NULL, w = "w_base")
summary(calibweights1)
#> Error in summary.ipf(calibweights1) : object 'extraExport' not found
```

https://github.com/statistikat/surveysd/blob/352290ea6b72e6ac12fbc65062e5413f68766118/R/summary.ipf.R#L100

This is likely a regression from https://github.com/statistikat/surveysd/pull/32/commits/925461f1a942f054be12d67068c0db59d3097bfe, where the named argument `extraExport` was replaced by an ellipsis (`...`).

File: R/summary.ipf.R [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      j <- ((i-1)*5)+1
+      names(output5)[(j):(j+4)] <- c(paste0("conH_",i),paste0("conH_",i,"_adjusted"),paste0("conH_",i,"_original"),
+                                     paste0("conH_",i,"_rel_diff_original"),paste0("conH_",i,"_rel_diff_calib"))
+      output5[[j]] <- as.data.table(av$conH[[i]]) #conH_i
+      output5[[j+1]] <- as.data.table(av$conH_adj[[i]]) #conH_i_adjusted
+      output5[[j+2]] <- as.data.table(xtabs(formHBase[[i]],data=x)) #conH_i_original
+      output5[[j+3]] <- as.data.table(round(100*(av$conH[[i]]-xtabs(formHBase[[i]],data=x))/av$conH[[i]],2)) #conH_i_rel_diff_original
+      output5[[j+4]] <- as.data.table(round(100*(av$conH[[i]]-av$conH_adj[[i]])/av$conH[[i]],2)) #conH_i__rel_diff_calib
+
+      output5[[i]] <- output55
```
Comment (GregorDeCillia) [inline R/summary.ipf.R]: `output55` is defined as `NULL` further up and never redeclared?

**Pull request merged**